gnmi_server/pamAuth.go

```diff
@@ -46,32 +47,30 @@ func PAMAuthUser(u string, p string) error {
 	err := cred.PAMAuthenticate()
 	return err
 }
-func GetUserRoles(usr *user.User) ([]string, error) {
-	// Lookup Roles
-	gids, err := usr.GroupIds()
-	if err != nil {
-		return nil, err
-	}
-	roles := make([]string, len(gids))
-	for idx, gid := range gids {
-		group, err := user.LookupGroupId(gid)
-		if err != nil {
-			return nil, err
+func GetUserRoles(username string) ([]string, error) {
+	// Lookup Roles using Dbus
+	var cmd string
+	cmd = "user_auth_mgmt.retrieve_user_roles"
+	host_output := transformer.HostQuery(cmd, username)
+	if host_output.Err != nil {
+		glog.Errorf("System user roles host query failed")
+		return nil,errors.New("Failed to retrieve user roles")
+	} else {
+		if val, _ := host_output.Body[0].(int32); val == 0 {
+			glog.Infof("Roles retrieved from host")
+			roles := strings.Split(host_output.Body[1].(string), ",")
```

Comment:
Please do not ignore errors. If `Body[1]` is not a string then the program will panic. Even if this not normally/today possible it might change in the future.